specfile/specfile.py

```diff
@@ -52,7 +64,8 @@ class Specfile:
 
     def __init__(
         self,
-        path: Union[Path, str],
+        path: Optional[Union[Path, str]] = None,
+        stream: Optional[StringIO] = None,
```

Comment:
I would probably use [ `file`](https://docs.python.org/3/glossary.html#term-file-object) as the argument name rather than `stream`, I think it's a bit more self-explanatory. Its meaning should definitely be documented in the docstring. The type should be `IOBase`, to cover all possibilities. There should be another argument of type `str` that, if specified, will create a `StringIO` instance and assign it to the `_stream` attribute (or perhaps `_file` to keep the naming consistent).